android/src/main/kotlin/com/google/maps/flutter/navigation/AndroidAutoBaseScreen.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.maps.flutter.navigation
+
+import android.app.Presentation
+import android.graphics.Point
+import android.hardware.display.DisplayManager
+import android.hardware.display.VirtualDisplay
+import androidx.car.app.AppManager
+import androidx.car.app.CarContext
+import androidx.car.app.Screen
+import androidx.car.app.SurfaceCallback
+import androidx.car.app.SurfaceContainer
+import androidx.car.app.model.Action
+import androidx.car.app.model.ActionStrip
+import androidx.car.app.model.Template
+import androidx.car.app.navigation.model.NavigationTemplate
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.GoogleMapOptions
+import com.google.android.libraries.navigation.NavigationViewForAuto
+
+open class AndroidAutoBaseScreen(carContext: CarContext) : Screen(carContext), SurfaceCallback {
+  private val VIRTUAL_DISPLAY_NAME = "AndroidAutoNavScreen"
+  private var mVirtualDisplay: VirtualDisplay? = null
+  private var mPresentation: Presentation? = null
+  private var mNavigationView: NavigationViewForAuto? = null
+  private var mAutoMapView: GoogleMapsAutoMapView? = null
+  var mGoogleMap: GoogleMap? = null
+
+  init {
+    initializeSurfaceCallback()
+  }
+
+  private fun initializeSurfaceCallback() {
+    carContext.getCarService(AppManager::class.java).setSurfaceCallback(this)
+  }
+
+  private fun isSurfaceReady(surfaceContainer: SurfaceContainer): Boolean {
+    return surfaceContainer.surface != null &&
+      surfaceContainer.dpi != 0 &&
+      surfaceContainer.height != 0 &&
+      surfaceContainer.width != 0
+  }
+
+  override fun onSurfaceAvailable(surfaceContainer: SurfaceContainer) {
+    super.onSurfaceAvailable(surfaceContainer)
+    if (!isSurfaceReady(surfaceContainer)) {
+      return
+    }
+    mVirtualDisplay =
+      carContext
+        .getSystemService(DisplayManager::class.java)
+        .createVirtualDisplay(
+          VIRTUAL_DISPLAY_NAME,
+          surfaceContainer.width,
+          surfaceContainer.height,
+          surfaceContainer.dpi,
+          surfaceContainer.surface,
+          DisplayManager.VIRTUAL_DISPLAY_FLAG_OWN_CONTENT_ONLY,
+        )
+    mPresentation = Presentation(carContext, mVirtualDisplay!!.display)
```

Comment:
Instead of `mVirtualDisplay!!` would it be better to return if `mVirtualDisplay` is null, instead of causing possible exception.
Same with mNavigationView in next line?

Reply:
Ah these are Java functions. I thought if the function returns non-optional it's guaranteed the be initialized. I'll fix it.

Reply:
Yeah. I think these are from the automatic java to kotlin conversion. I'll clean it up.

Reply:
Done.